app/src/main/java/com/github/codetanzania/ui/activity/ReportIssueActivity.java

```diff
@@ -439,4 +458,20 @@ public void onRemovePreviewItemClicked() {
             this.optionalBitmapAttachment = null;
         }
     }
+
+    @Override public void onBackPressed() {
+        if (mSubmissionTicket == null) {
+            // Display an alert dialog. User is about to close the issue without posting
+            new AlertDialog.Builder(this)
+                .setMessage(R.string.text_confirm_cancel_new_issue)
+                .setPositiveButton(R.string.action_confirm_exit, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        finishWithResult();
+                    }
+                })
+                .setPositiveButton(R.string.text_cancel, null)
+                .show();

```

Comment:
This should have a Positive Button ("Yes") and a Negative Button ("No"). When I click yes, I should exit. When I click no, I should stay in the issue creation fragment. When there was a server error, and an issue would not submit, I was stuck in an endless loop and had to exit the app.